pkg/build/gobuild.go

```diff
@@ -928,8 +932,12 @@ func (g *gobuild) buildAll(ctx context.Context, ref string, baseRef name.Referen
 	// Annotate the index with base image information, if the index is an OCI image index.
 	// (Docker manifest lists don't support annotations)
 	if baseType == types.OCIImageIndex {
+		var baseName string
+		if _, ok := baseRef.(name.Tag); ok {
+			baseName = baseRef.Name()
+		}
 		idx = mutate.Annotations(idx, map[string]string{
-			specsv1.AnnotationBaseImageName:   baseRef.Name(),
+			specsv1.AnnotationBaseImageName:   baseName,
```

Comment:
Won't this put an empty string in the map?

Reply:
Yeah, I'm doing the same thing in https://github.com/google/go-containerregistry/pull/1098 but I can make both just omit the key entirely.

Reply:
done